### PR TITLE
Restrict set_exchange_rate oracle function. Allow only for tests and benchmarks.

### PR DESCRIPTION
**Pull request by cr4pt0**

### CODE REVIEW

File: pallets/oracle/src/lib.rs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 //! # Oracle Pallet
 //! Based on the [specification](https://spec.interlay.io/spec/oracle.html).
 
-#![deny(warnings)]
+#![warn(warnings)]
```
Comment (ebma) [inline pallets/oracle/src/lib.rs:4]: Is this change necessary or did you forget to change it back?

**Review by ebma — APPROVED** [review]
Looks good to me overall. I checked and the tests are all still working. Goob job 👍 

We can merge this once you address my other comment.

File: pallets/oracle/src/lib.rs (hunk 1 above)
Comment (cr4pt0) [reply pallets/oracle/src/lib.rs:4]: @ebma 
I did it because the test failed because of `unused import: `DispatchResult``

Should I fix it or we can leave  it like this?

Comment (ebma) [reply pallets/oracle/src/lib.rs:4]: Can't we just remove the unused import then?

Comment (cr4pt0) [reply pallets/oracle/src/lib.rs:4]: No. Because `DispatchResult` for other code when there is no feature `testing-utils`

Comment (ebma) [reply pallets/oracle/src/lib.rs]: I fixed it by conditionally importing it.
